src/gui/application.cpp

```diff
@@ -646,31 +549,106 @@ static void displayHelpText(const QString &t, std::ostream &stream = std::cout)
 #endif
 }
 
-void Application::showHelp()
+void Application::parseOptions(const QStringList &arguments)
 {
-    setHelp();
-    QString helpText;
-    QTextStream stream(&helpText);
-    stream << _theme->appName()
-           << QLatin1String(" version ")
-           << _theme->version() << endl;
+    QCommandLineParser parser;
+
+    QString descriptionText;
+    QTextStream descriptionTextStream(&descriptionText);
 
-    stream << QLatin1String("File synchronisation desktop utility.") << endl
-           << endl
-           << optionsC();
+    descriptionTextStream << _theme->appName() << QStringLiteral(" version ") << _theme->version() << endl;
+    descriptionTextStream << QStringLiteral("File synchronization desktop utility.");
 
     if (_theme->appName() == QLatin1String("ownCloud")) {
-        stream << endl
-               << endl
-               << "For more information, see http://www.owncloud.org";
+        descriptionTextStream << endl
+                              << endl
+                              << "For more information, see http://www.owncloud.org";
     }
 
-    displayHelpText(helpText);
-}
+    parser.setApplicationDescription(descriptionText);
+
+    auto helpOption = parser.addHelpOption();
+    auto versionOption = parser.addVersionOption();
+
+    // this little snippet saves a few lines below
+    auto addOption = [&parser](const QCommandLineOption &option) {
+        parser.addOption(option);
+        return option;
+    };
+
+    auto showSettingsOption = addOption({ { "s", "showsettings" }, "Show the settings dialog while starting." });
+    auto quitInstanceOption = addOption({ { "q", "quit" }, "Quit the running instance." });
+    auto logFileOption = addOption({ "logfile", "Write log to file (use - to write to stdout).", "filename" });
+    auto logDirOption = addOption({ "logdir", "Write each sync log output in a new file in folder.", "name" });
+    auto logExpireOption = addOption({ "logexpire", "Remove logs older than <hours> hours (to be used with --logdir).", "hours" });
+    auto logFlushOption = addOption({ "logflush", "Flush the log file after every write." });
+    auto logDebugOption = addOption({ "logdebug", "Output debug-level messages in the log." });
+    auto logWindowOption = addOption({ "logwindow", "//TODO//" });
+    auto languageOption = addOption({ "language", "Override UI language.", "language" });
+    auto confDirOption = addOption({ "confdir", "Use the given configuration folder.", "dirname" });
+    auto debugOption = addOption({ "debug", "Enable debug mode." });
+
+    // virtual file system parameters (optional)
+    parser.addPositionalArgument("vfs file", "Virtual file system file to be opened (optional).", { "[<vfs file>]" });
```

Comment:
Hmm sadly I don't think the commandlineparser allows to hide options, as this is never meant to be used in a direct invocation but only by the os.

Reply:
I don't think there's a point in hiding it, really. How many people will ever look at this help anyway? It's for advanced users. We could extend the description if you want to.

Reply:
Hiding also doesn't seem to be possible with the standard Qt parser.